### PR TITLE
Add spec for shared-history room keys (MSC3061)

### PR DESCRIPTION
**Pull request by uhoreg**
spec PR for https://github.com/matrix-org/matrix-spec-proposals/pull/3061

Preview: https://pr1655--matrix-spec-previews.netlify.app

### CODE REVIEW

**Review by poljar — COMMENTED** [review]
Should we push this to the spec, considering the problems which were subsequently discovered with this MSC3061?

**Review by richvdh — CHANGES_REQUESTED** [review]
Pending a decision about wtf we are doing here

**Comment by richvdh** [comment]
Apparently there are security concerns, specifically around sharing keys for past messages. We'll disclose more details in a few weeks, and then we can decide how best to proceed here.

File: content/client-server-api/modules/end_to_end_encryption.md [hunk 1]
```diff
@@ -1259,14 +1261,50 @@ of the same user, and should only request and accept forwarded keys from
...
+other devices. For example, when inviting a user to an encrypted room, the
+inviter may choose to share only the keys with `shared_history` set to `true`.
+
+{{% boxes/note %}}
+When sharing keys for old messages, clients must not blindly trust the current
+state as reported by the homeserver: clients should not assume that users who
+have `m.room.member` events with `membership: "join"` are legitimately in the
+room, as such events could be spoofed by the homeserver.  Clients should use
+other means of ensuring that a user is actually in the room before sharing keys
+for old messages with them.  For example, clients can share keys for old
```
Comment (Johennes) [inline content/client-server-api/modules/end_to_end_encryption.md:1295]:
```suggestion
for old messages with them. For example, clients can share keys for old
```

**Review by turt2live — COMMENTED** [review]
clearing team review request while awaiting security impact

**Comment by murlock1000** [comment]
With the recent migration of element clients to the rust SDK this introduced a regression of the spec. Any updates on this @turt2live ?

**Comment by dkasak** [comment]
The reason this was blocked was disclosed in https://matrix.org/blog/2024/10/security-disclosure-matrix-js-sdk-and-matrix-react-sdk/ and the associated advisories. The plan forward is outlined in [this section](https://matrix.org/blog/2024/10/security-disclosure-matrix-js-sdk-and-matrix-react-sdk/#proposed-specification-changes).

**Review by turt2live — CHANGES_REQUESTED** [review]
Formalizing "proposed spec changes" at https://matrix.org/blog/2024/10/security-disclosure-matrix-js-sdk-and-matrix-react-sdk/#proposed-specification-changes as requiring action. That future MSC will have impact on this spec PR's text.

**Comment by richvdh** [comment]
MSC3061 has been un-merged, so we're going to abandon this for now.

**Pull request closed without merging**